test/integration/formatter_test.py

```diff
@@ -0,0 +1,148 @@
+import pytest
+from inspect import cleandoc
+
+
+@pytest.mark.format
+@pytest.mark.parametrize("value,expected", [
+    (None, None),
+    (1, "1"),
+    (1.0, "1.0"),
+])
+def test_no_conversion_specified(connection, value, expected):
+    params = {"value": value}
+    query = "SELECT {value};"
+    actual = connection.execute(query, params).fetchval()
+
+    assert expected == actual
+
+
+@pytest.mark.format
+def test_safe_quote_conversion(connection, schema):
+    params = {"table": (schema, "USERS")}
+    query = cleandoc("""
+        SELECT user_name FROM {table!q}
+        ORDER BY user_name DESC
+        LIMIT 5;
+    """)
+    expected = [
+        'Zoe Odom',
+        'Zoe Merritt',
+        'Zoe Lawrence',
+        'Zoe Fisher',
+        'Zachary Wood'
+    ]
+    actual = connection.execute(query, params).fetchcol()
+
+    assert expected == actual
+
+
+@pytest.mark.format
+@pytest.mark.parametrize("value", [
+    True, 'True', "TRUE", 'true'
+])
+def test_string_conversion(connection, value):
+    params = {"is_female": value}
+    query = cleandoc("""
+        SELECT user_name FROM USERS
+        WHERE is_female IS {is_female!r}
+        ORDER BY user_name DESC
+        LIMIT 5;
+    """)
+    expected = [
+        'Zoe Odom',
+        'Zoe Merritt',
+        'Zachary Williams',
+        'Zachary Torres',
+        'Zachary Thompson'
+    ]
+    actual = connection.execute(query, params).fetchcol()
+
+    assert expected == actual
+
+
+@pytest.mark.format
+def test_safe_identifier_conversion(connection):
+    params = {"table": "USERS"}
```

Comment:
Probably similar as for the quoting, a lowercase table name would be good

Reply:
works with lower case -> changed.